src/pydot/dot_parser.py

```diff
@@ -113,57 +113,21 @@ def parseImpl(
         )
 
 
-def push_top_graph_stmt(
-    s: str, loc: int, toks: ParseResults
-) -> list[pydot.core.Dot] | pydot.core.Dot:
-    attrs = {}
+def push_top_graph_stmt(toks: ParseResults) -> list[pydot.core.Dot]:
     top_graphs = []
-    g: pydot.core.Dot = None  # type: ignore
-
-    for element in toks:
-        if (
-            isinstance(element, (ParseResults, tuple, list))
-            and len(element) == 1
-            and isinstance(element[0], str)
-        ):
-            element = element[0]
-
-        if element == "strict":
-            attrs["strict"] = True
-
-        elif element in ["graph", "digraph"]:
-            attrs = {}
-
-            g = pydot.core.Dot(graph_type=element, **attrs)
-            attrs["type"] = element
 
-            top_graphs.append(g)
+    for result in toks.graphs:
+        assert isinstance(result, ParseResults)
+        gtype = result.gtype
+        strict = "strict" in result
+        id_ = result.id or ""
```

Comment:
This `or` here, OTOH, I'm now realizing is dumb. Named results in `ParseResults` objects _default_ to returning `''` if they haven't captured any data. They'll never be `None`, as I was apparently assuming might happen when I wrote this.

I mean, it's harmless as such things go; if `result.id` is `''` the `or` will set `id_` to the same thing it would've been set to anyway. `¯\_(ツ)_/¯`

Reply:
Fixed it anyway.